docs/proposals/job.md

```diff
@@ -0,0 +1,135 @@
+Additionally this new controller should take over pod management logic we currently
+have in certain OpenShift controllers, namely build controller.
+
+
+## Implementation
+Job controller, is similar to replication controller in that they manage pods
+representing a `Job`.  This implies they will follow the same controller framework
+that replication controllers already defined.  The biggest difference between
+`Job` and `ReplicationController` objects is the `RestartPolicy` which is required
+to always take value of `RestartPolicyNever` or `RestartOnFailure`.  The main
+responsibility of a job controller will be keeping that the number of pods will
+match the number specified in the job specification in such way that the job
+should finish its calculation.
+
+
+The new `Job` object will have the following content:
+
+```go
+// Job represents the configuration of a single job.
+type Job struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job.
+    Spec JobSpec
+
+    // Status is a structure describing current status of a job.
+    Status JobStatus
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []Job
+}
+```
+
+`JobSpec` structure is defined to contain all the information how the actual job execution
+will look like.
+
+```go
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+    // PodCount specifies the desired number of pods the job should be run with.
+    PodCount int
+
+    // Selector is a label query over pods that should match the pod count.
+    Selector map[string]string
+
+    // Spec is the object that describes the pod that will be created when
+    // executing a job.
+    Spec PodSpec
+}
+```
+
+`JobStatus` structure is defined to contain informations about pods executing
+specified job.  The structure holds information about pods currently executing
+the job.
+
+```go
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+    // Executions holds a detailed information about each of the pods running a job.
+    Executions []JobExec
+
+    // Completions is the number of pods successfully completed their job.
+    Completions int
+
+}
+
+// JobExec represents the current state of a single execution of a Job.
+type JobExec struct {
+    // CreatedAt represents time when the job execution was created
+    CreatedAt util.Time
+
+    // StartedAt represents time when the job execution was started
+    StartedAt util.Time
+
+    // CompletedAt represents time when the job execution was completed
+    CompletedAt util.Time
+
+    // Phase represents the point in the job execution lifecycle.
+    Phase JobExecPhase
+
+    // Tag is added in labels of pod(s) created for this job execution.  It allows
+    // job object to safely group/track all pods started for one given job execution.
+    Tag util.UID
```

Comment:
I'm seeing this similar to what we have with deployments or builds, a Job represents definition (intention), job execution is actual run, thus the ability to have more than one. Where each is executed on demand, similarly to builds.